src/App.tsx

```diff
@@ -1,223 +1,214 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useRef, useState, useCallback, useMemo,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoList } from './components/TodoList';
+import { FilterTypes } from './types/FilterTypes';
+import { Footer } from './components/Footer';
+import { TodoField } from './components/TodoField';
+
+import {
+  getTodos,
+  addTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorTypes } from './types/ErrorType';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<ErrorTypes | null>(null);
+  const [filterType, setFilterType] = useState<FilterTypes>(FilterTypes.all);
+  const [isAdding, setIsAdding] = useState(false);
+  const [todoToAdd, setTodoToAdd] = useState('');
+  const [currTodo, setCurrTodo] = useState(0);
+  const [isEditing, setisEditing] = useState(false);
+
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const filteredTodos = todos.filter(todo => {
+    switch (filterType) {
+      case FilterTypes.active:
+        return !todo.completed;
+
+      case FilterTypes.completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  const todoNumber = (items: Todo[]) => items.filter(todo => !todo.completed);
+  const completedTodo = (items: Todo[]) => items.filter(todo => todo.completed);
+
+  const notCompleted = useMemo(() => (
+    todoNumber(todos)
+  ), [todos, filterType]);
+
+  const completed = useMemo(() => (
+    completedTodo(todos)
+  ), [todos, filterType]);
```

Comment:
check names of variables
```suggestion
  const todoNumber = (items: Todo[]) => items.filter(todo => !todo.completed);
  const completedTodo = (items: Todo[]) => items.filter(todo => todo.completed);

  const notCompletedTodos = useMemo(() => (
    todoNumber(todos)
  ), [todos, filterType]);

  const completedTodos = useMemo(() => (
    completedTodo(todos)
  ), [todos, filterType]);
```